File: PerfectionOrNot/PerfectionOrNot.java
import java.util.Scanner;

public class PerfectionOrNot {
	
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		
		int num = scan.nextInt();
		
		int sum = 0;
		
		for(int i = 1; i < num; i++) if(num % i == 0) sum += i;
		
		System.out.print((sum == num) ? "YES" : "NO");
	}
}

File: PerfectionOrNot/PerfectionOrNot.c
#include <stdio.h>

int div[42];
int last=0;
void finddivisors(int n){
    int i;
    for(i=1;i<=n/2;i++){
        if(n%i==0){
            div[last++]=i;
        }
    }
}

int main(){
    int sum=0;
    int num;
    scanf("%d",&num);
    finddivisors(num);
    int i;
    for(i=0;i<last;i++){
        sum += div[i];
    }
    if(sum==num){
        printf("YES\n");
    }
    else {
        printf("NO\n");
    }
}
